### ISSUE TITLE
SiteMap functionality missing Settings is missing from 3.4.3 release

### ISSUE THREAD

**Issue by leigh-pointer** [oqtane/oqtane.framework#2787]
The Fix for #2718 SiteMap functionality missing Settings is missing from 3.4.3 release - Source Code oqtane.framework-3.4.3 and from the compiled package

**Comment by thabaum** [comment]
```
        public IActionResult OnGet()
        {
            var sitemap = new List<Sitemap>();

            // build site map
            var moduleDefinitions = _moduleDefinitions.GetModuleDefinitions(_alias.SiteId).ToList();
            var pageModules = _pageModules.GetPageModules(_alias.SiteId);
                        foreach (var page in _pages.GetPages(_alias.SiteId))
            {
                if (_userPermissions.IsAuthorized(null, PermissionNames.View, page.PermissionList) && page.IsNavigation)
                {
                    sitemap.Add(new Sitemap { Url = _alias.Protocol + _alias.Name + Utilities.NavigateUrl(_alias.Path, page.Path, ""), ModifiedOn = DateTime.UtcNow });

                    foreach (var pageModule in pageModules.Where(item => item.PageId == page.PageId))
                    {
                        if (_userPermissions.IsAuthorized(null, PermissionNames.View, pageModule.Module.PermissionList))
                        {
                            var moduleDefinition = moduleDefinitions.Where(item => item.ModuleDefinitionName == pageModule.Module.ModuleDefinitionName).FirstOrDefault();
                            if (moduleDefinition != null && moduleDefinition.ServerManagerType != "")
                            {
                                Type moduletype = Type.GetType(moduleDefinition.ServerManagerType);
                                if (moduletype != null && moduletype.GetInterface("ISitemap") != null)
                                {
                                    try
                                    {
                                        var moduleobject = ActivatorUtilities.CreateInstance(_serviceProvider, moduletype);
                                        var urls = ((ISitemap)moduleobject).GetUrls(_alias.Path, page.Path, pageModule.Module);
                                        foreach (var url in urls)
                                        {
                                            sitemap.Add(new Sitemap { Url = _alias.Protocol + _alias.Name + url.Url, ModifiedOn = DateTime.UtcNow });
                                        }
                                    }
                                    catch (Exception ex)
                                    {
                                        _logger.Log(LogLevel.Error, this, LogFunction.Other, ex, "Error Retrieving SiteMap For {Name} Module", moduleDefinition.Name);
                                    }
                                }
                            }
                        }
                    }
                }
            }
```

It looks like this line didnt make it for some reason in my final version so yes this is from my last PR.

`                                        pageModule.Module.Settings = _settings.GetSettings(EntityNames.Module, pageModule.ModuleId).ToDictionary(x => x.SettingName, x => x.SettingValue);`

is missing here:

```

                                    try
                                    {
                                        pageModule.Module.Settings = _settings.GetSettings(EntityNames.Module, pageModule.ModuleId).ToDictionary(x => x.SettingName, x => x.SettingValue);
                                        var moduleobject = ActivatorUtilities.CreateInstance(_serviceProvider, moduletype);
                                        var urls = ((ISitemap)moduleobject).GetUrls(_alias.Path, page.Path, pageModule.Module);
                                        foreach (var url in urls)
                                        {
                                            sitemap.Add(new Sitemap { Url = _alias.Protocol + _alias.Name + url.Url, ModifiedOn = DateTime.UtcNow });
                                        }
                                    }
```

**Comment by thabaum** [comment]
Well i have the settings DI in the PR, so not sure how but I bet it was a master copy i was working at some point while working in the GetOn method and didnt realize it and I was trying to be careful here to be sure to not foul up that PR of yours.  I take credit for this one as it would be easy to miss.

I will delete all my local copies going forward to be sure things reflect latest changes.  I have many copies going so trying to get organized again after a long break from this arena.

Thanks for the tip @leigh-pointer it was an honest mistake as I really wanted your settings PR included not to be touched and included in my PR as I am aware of PR's working on the same file having an issue like this if not careful.  I will be double careful and check 4 times next time I am working on top of another PR in upcoming releases.

**Comment by thabaum** [comment]
@leigh-pointer do you want to create the fix PR for this?  I will if not later today?  If you do go ahead put it up I will make up for this in other areas...

**Comment by sbwalker** [comment]
@leigh-pointer my apologies... I reviewed the #2747 PR but there were so many changes that were unrelated to the original issue it was intended to fix, it was quite difficult to validate.

**Comment by thabaum** [comment]
@sbwalker I had this covered as your scapegoat, i seen it and figured it was due to me somehow.

**Comment by sbwalker** [comment]
resolved in #2797
